### PR TITLE
upgrade java agent attacher to 1.50.0

### PR DESCRIPTION
**Pull request by jackshirazi**
## Motivation/summary

The Java attacher version needs to be upgraded to 1.50.0. The PGP sig is taken from the [agent attacher source](https://github.com/elastic/apm-agent-java/blob/main/apm-agent-attach-cli/src/main/resources/pub_key_8AB554FD8F207067.asc)

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

Check the deployment uses java agent 1.50.0 artifacts

### CODE REVIEW

**Comment by jackshirazi** [comment]
run docs-build

File: packaging.mk [hunk 1]
```diff
@@ -1,228 +1,228 @@
-##############################################################################
```
[the remark this answers is not in the document]
Comment (jackshirazi) [reply packaging.mk:1]: try viewing with whitespace turned off. Git seems to have got really confused with me editing on the PC (it doesn't normally, and I tried switching to the ubuntu subsystem but it still did this)

[the remark this answers is not in the document]
Comment (trentm) [reply packaging.mk:1]: This file as Windows (CRLF) line endings now, hence the huge diff.  I could push a commit changing it back to Unix (LF) line endings if you like.

I was going to push a commit to fix the line-endings, but was denied  :)

```
% git remote get-url jackshirazi
git@github.com:jackshirazi/apm-server.git
% git push jackshirazi upgrade-attacher-to-1-5-0
ERROR: Permission to jackshirazi/apm-server.git denied to trentm.
...
```

[the remark this answers is not in the document]
Comment (trentm) [reply packaging.mk:1]: If it helps, here is a single-file, no-deps Python script for querying and change the line-endings of files: https://github.com/trentm/eol/blob/master/lib/eol.py

```
% eol $(git diff --name-only main...)
eol: info: changelogs/head.asciidoc: Unix (LF)
eol: info: packaging.mk: Windows (CRLF)
eol: info: packaging/apm-agent-java-public-key.asc: Unix (LF)

% eol -c unix packaging.mk
eol: info: converted `packaging.mk' to LF EOLs
```

[the remark this answers is not in the document]
Comment (jackshirazi) [reply packaging.mk]: I ran dos2unix, looks like it worked

**Comment by jackshirazi** [comment]
closing this for https://github.com/elastic/apm-server/pull/13326

**Pull request closed without merging**